src/formats/PDB.cpp

```diff
@@ -625,18 +628,14 @@ Record get_record(string_view line) {
     }
 }
 
-static std::string to_pdb_index(uint64_t i) {
-    auto id = i + 1;
-
-    if (id > 99999) {
-        if (id == 99999) {
-            // Only warn once for this
-            warning("PDB writer", "too many atoms, removing atomic id bigger than 100000");
-        }
-        return "*****";
-    } else {
-        return std::to_string(i + 1);
+static std::string to_pdb_index(int64_t value, uint64_t width) {
+    if (value == 10000 + 2 * 26 * static_cast<int64_t>(std::pow(36, width - 1) + 0.5)) {
+        auto type = width == 5 ?
+            "atom" : "residue";
+        warning("PDB writer", "the value for a {} serial/id is too large, using '*' instead", type);
```

Comment:
Is this missing a `return "****";`?

Reply:
No, but the code is unclear - I'll refactor.